src/attributes.js

```diff
@@ -20,7 +20,14 @@ export default function attributes({id, ripple, primary, primaryDark, active, di
 	if(primary) attr.class.push('mdl-color--primary');
 	if(primaryDark) attr.class.push('mdl-color--primary-dark');
 	if(shadow) attr.class.push(`mdl-shadow--${shadow}dp`);
-	if(arguments[0].class) attr.class.push(arguments[0].class);
+	if(color) attr.class.push(`mdl-color--${color}`);
+	if(textColor) attr.class.push(`mdl-color-text--${textColor}`);
+
+	let classArg = arguments[0].class;
+	if(classArg) {
+		if(typeof(classArg) === 'string') attr.class.push(classArg);
```

Comment:
If else please with braces. Should really add the linting rules for that at some point.